### PR TITLE
Fixes #1725

### LINKED ISSUE #1725: Unittest Autocoder Fails on Namespace
| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|   |
---
## Problem Description

`TestMain.cpp` is created with a stub unit test that hardcodes the namespace `Ref` rather than detecting it from the model.

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Corrects the cheetah template to fix #1725.

### CODE REVIEW

**Comment by LeStarch** [comment]
New output in Svc::Framer:

```
    Svc::Tester tester;
```

File: Autocoders/Python/src/fprime_ac/generators/templates/test/test_main.tmpl [hunk 1]
```diff
@@ -6,7 +6,11 @@
 
 #for $test_name, $method_name in $test_cases
 TEST(Nominal, ${test_name}) {
-    Ref::Tester tester;
+    #if $namespace_list != None
+    #echo "::".join($namespace_list)#::Tester tester;
+    #else
+    Tester tester;
```
Comment (thomas-bc) [inline Autocoders/Python/src/fprime_ac/generators/templates/test/test_main.tmpl:12]: Are there legitimate cases where the namespace_list could be empty/none? Not that the handling should be different, just wondering.

**Review by thomas-bc — APPROVED** [review]
I also tested with the Imu in SystemReference, looks good!

File: Autocoders/Python/src/fprime_ac/generators/templates/test/test_main.tmpl (hunk 1 above)
Comment (LeStarch) [reply Autocoders/Python/src/fprime_ac/generators/templates/test/test_main.tmpl:12]: I don't believe it is possible, but it is correct to have it implemented in the case I am mistaken.

**Pull request merged**